rocks-bin/src/rockspec/write_rockspec.rs

```diff
@@ -0,0 +1,319 @@
+                            continue;
+                        }
+                    }
+                }
+                Err(ReadlineError::Interrupted) => {
+                    break Err(eyre!("Ctrl-C pressed, exiting..."));
+                }
+                Err(ReadlineError::Eof) => {
+                    break Err(eyre!("Ctrl-D pressed, exiting..."));
+                }
+                Err(err) => break Err(err.into()),
+            }
+        }
+    };
+}
+
+// General notes and ideas:
+// - Should we require the user to create a "project" in order to use this command?
+// - Automatically detect build type by inspecting the current repo (is there a Cargo.toml? is
+//   there something that tells us it's a lua project?).
+// - If repo metadata couldn't be fetched then don't error, simply don't provide default values.
+
+#[derive(Args)]
+pub struct WriteRockspec {
+    /// The directory of the rock.
+    directory: Option<PathBuf>,
+
+    /// The name to give to the rock.
+    #[arg(long)]
+    name: Option<String>,
+
+    /// The description of the rock.
+    #[arg(long)]
+    description: Option<String>,
+
+    /// The license of the rock. Generic license names will be inferred.
+    #[arg(long, value_parser = parse_license_wrapper)]
+    license: Option<std::option::Option<LicenseId>>, // Note: full qualified name required, see https://github.com/clap-rs/clap/issues/4626
+
+    /// The maintainer of this rock. Does not have to be the code author.
+    #[arg(long)]
+    maintainer: Option<String>,
+
+    /// A comma-separated list of labels to apply to this rock.
+    #[arg(long, value_parser = parse_list_wrapper)]
+    labels: Option<std::vec::Vec<String>>, // Note: full qualified name required, see https://github.com/clap-rs/clap/issues/4626
+
+    /// A version constraint on the required Lua version for this rock.
+    /// Examples: ">=5.1", "5.1"
+    #[arg(long, value_parser = parse_version_wrapper)]
+    lua_versions: Option<LuaDependency>,
+}
+
+fn parse_license_wrapper(s: &str) -> std::result::Result<Option<LicenseId>, String> {
+    Ok(parse_license(s.to_string()).map_err(|err| err.to_string())?)
+}
+
+fn parse_version_wrapper(s: &str) -> std::result::Result<LuaDependency, String> {
+    Ok(parse_version(s.to_string()).map_err(|err| err.to_string())?)
+}
+
+fn parse_list_wrapper(s: &str) -> std::result::Result<Vec<String>, String> {
+    Ok(parse_list(s.to_string()).map_err(|err| err.to_string())?)
+}
+
+fn identity(input: String) -> Result<String> {
+    Ok(input)
+}
+
+fn parse_list(input: String) -> Result<Vec<String>> {
+    if let Some((pos, char)) = input
+        .chars()
+        .find_position(|&c| c != '-' && c != '_' && c != ',' && c.is_ascii_punctuation())
+    {
+        Err(eyre!("Unexpected punctuation '{}' found at column {}. Lists are comma separated but names should not contain punctuation!", char, pos))
+    } else {
+        Ok(input.split(",").map(|str| str.trim().to_string()).collect())
```

Comment:
clippy is failing on this

```suggestion
        Ok(input.split(',').map(|str| str.trim().to_string()).collect())
```